crates/circuits/poseidon2-air/src/air.rs

```diff
@@ -1,16 +1,510 @@
+use core::{
+    borrow::{Borrow, BorrowMut},
+    marker::PhantomData,
+    mem::size_of,
+};
+use std::{string, vec};
+
+use openvm_columns_core::FlattenFieldsHelper;
 use openvm_stark_backend::{
-    p3_air::{Air, AirBuilder, BaseAir},
-    p3_field::Field,
-    rap::{BaseAirWithPublicValues, PartitionedBaseAir},
+    p3_field::{Field, FieldAlgebra},
+    p3_matrix::Matrix,
+    rap::{Air, AirBuilder, BaseAir, BaseAirWithPublicValues, PartitionedBaseAir},
 };
-use p3_poseidon2_air::{Poseidon2Air, Poseidon2Cols};
+use p3_poseidon2::GenericPoseidon2LinearLayers;
+use rand::{distributions::Standard, prelude::Distribution, Rng};
 
 use super::{
     BABY_BEAR_POSEIDON2_HALF_FULL_ROUNDS, BABY_BEAR_POSEIDON2_PARTIAL_ROUNDS,
     BABY_BEAR_POSEIDON2_SBOX_DEGREE, POSEIDON2_WIDTH,
 };
 use crate::{BabyBearPoseidon2LinearLayers, Plonky3RoundConstants};
 
+// Round constants for Poseidon2, in a format that's convenient for the AIR.
```

Comment:
Both this code block, as well as the one added in `Keccak-Air` come from `Plonky3` and are the result of importing `BaseAir` into rap